podip_vip.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: podip-vip-mappings
+  namespace: default
+data:
+  10.244.2.2: 192.168.122.222
```

Comment:
Thank you for your feedback. 

> Pod IPs are usually dynamic unless using a specific ClusterIP, so I'm not sure the value of hardcoding them here.

To dynamically update current Pod IP, I meant to introduce egress_mgr.sh, as an example.
This implementation could be replaced with operator to watch the change in Pod, instead of polling by using shell script. (This should reduce the lag.) And it might be better to define map to service, instead of mapping to pod directly, as you suggested.
(podip-vip-mappings configmap should be automatically created/updated from above tools, in a real usecase. And it won't be created manually. Sorry for making you confuse. I'll update README and delete pod_ip.yaml)

The basic idea is that kube-egress can keep simple in a way that it's role is just forwarding packets from Pod IP as packets from VIP, if such an external component keep updating mappings for Pod and Pod IP. So, the each script's role is like below:

- egress_mgr.sh: regularly read user's configuration for pod and VIP mapping, check the current Pod IP, and update the configmap for pod IP and VIP mapping.
- kube-egress.sh: regularly read configmap for pod IP and VIP mapping and update iptables and routing table in all nodes.

Of course, kube-egress.sh can do the same work to egress_mgr.sh, but it won't be necessary to be done in all nodes, as it is running as daemonset.